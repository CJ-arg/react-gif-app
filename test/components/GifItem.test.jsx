import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"
import React from "react"

describe('pruebas en GIFTITEM', () => {
  const title = 'Clarian'
  const url = 'https: //clarian.com'
  test('should match snapshot', () => {

    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()

  });
  test('should show the image url and alt ', () => {
    render(<GifItem title={title} url={url} />)
    // expect(screen.getByRole('img').src).toBe(url)
    // expect(screen.getByRole('img').alt).toBe(title)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(alt)
  })
  test('should show a title in the component ', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })
})