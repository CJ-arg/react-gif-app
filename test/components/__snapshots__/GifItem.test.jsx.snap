// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GIFTITEM should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Clarian"
      src="https: //clarian.com"
    />
    <p>
       
      Clarian
       
    </p>
  </div>
</div>
`;
